devops/providers/azure-devops/templates/app_service_maven/examples/service_usage.md

```diff
@@ -0,0 +1,146 @@
+           ${{ if ne(variables['Build.SourceBranchName'], 'master') }}:
+             environments: ['devint']
+```
+
+Services will typically leverage the following common templates to configure their build and release stages. All configurations can be overridden:
+
+- `devops/providers/azure-devops/templates/app_service_maven/build-stage.yml`
+
+- `devops/providers/azure-devops/templates/app_service_maven/deploy-stages.yml`
+
+  **Required Configurations**
+
+  - `repositories` keyword mapping: This is the syntax needed to reference an external repository. The **Shared Maven Service Pipeline** templates live in the infrastructure repository and not in the service repository with the **Maven Service**.
+  - `template` attribute keyword: This is the syntax needed to use another `yaml` file as a template. The `yaml` files being referenced must be either the full path to the build stage `yaml` file or the deploy stages `yaml` file.
+  - `copyFileContents` or `copyFileContentsToFlatten` parameter: Either or both of these parameters must be satisifed with the paths that point to the files you want packaged up and passed as build artifacts.
+
+  The following key areas impact the variable groups that are required for the pipeline:
+
+  - Parameter which defines the `providers`. This controls which cloud provider the application will be deployed to. In this example, we use Azure.
+  - Stanza which defines the `environments`. This controls which environment the application will be deployed to. It should match the environments configured in the infrastructure pipeline. In the example shown here, the environments deployed will depend on whether or not the build has been triggered from the `master` branch. This enables PR builds to deploy only to `devint`.
+  - Stanza which defines  the `serviceBase`: This controls the name of the Maven Service being deployed. It should be unique for each Maven Service being deployed. In this example, we use `JavaHelloWorld` as a generic service application name.
+
+  **Optional Parameters**
+
+  - `mavenOptions` Use this parameter to pass additional values to your settings.xml file.
+  - `mavenGoal`: Use this parameter to pass in different build lifecycle commands
+  - `testingRootFolder`: Use this parameter to pass the path to your service's integration tests.
+  - `mavenPomFile`: 'Use this parameter to pass the path to your pom file.
+
+### Step 2: Configure the Azure DevOps Variable Groups
+
+Variable groups are named in a way that allows the **Shared Maven Service Pipeline** to look up rather or not the group belongs to a specific cloud provider and for which environment should the group be used for. The following table describes the variable groups required to support a service deployment:
+
+`Azure - Common`
+
+  This group depends on all the above mentioned variable groups. It also holds the majority of variables needed to deploy a **Maven Service**.
+
+  | name | value | description | sensitive? | source |
```

Comment:
Since this is a hello-world only, I'd imagine only the variables needed by the pipeline should live here. A lot of these are application specific. 

Can we reduce these down to the minimum needed for the *deployment* and assume that the application does not need any additional params?

Similar comment for other groups listed here

Reply:
I might have to think about this a bit more. I feel as though the variables in these groups serve as good examples for targeting pre-existing infrastructure in ADO. The `Azure Service Release - $SERVICE` variable groups is what ends up consuming most of these variables for deployment and integration testing. Integration test too much for a "helloworld"?

Reply:
Maybe they can be simplified into something smaller, such as a single variable that is needed and can be passed to the test? something like `SAMPLE_VARIABLE_FOR_INTEGRATION_TESTING` with a value of `FOO` so that its clear what it is

Reply:
Gotcha

Reply:
I think as long as we make it clear that which of the variables here are *examples* vs which are *necessary*, then we are OK

Reply:
Discussed on standup - I had misunderstood the context of the document. With the new understanding, this looks great!